### PR TITLE
Add markdown tests

### PR DESCRIPTION
**Pull request by YousefED**
- Extracted the customblock case from htmlconversions to "testCases" directory
- Added tests for markdown parsing and exporting
- Added tests for #226 
- Updated docs with updated function names

TODO: @matthewlipski would be nice if you can have an extra look at the snapshots

### CODE REVIEW

File: packages/core/src/api/exporters/markdown/__snapshots__/simpleImage/button/markdown.md [hunk 1]
```diff
@@ -0,0 +1 @@
+![placeholder]()
```
[the remark this answers is not in the document]
Comment (YousefED) [reply packages/core/src/api/exporters/markdown/__snapshots__/simpleImage/button/markdown.md:1]: Good spot. Another approach could be to make the default ExternalHTML export remove any hidden or invisible elements, wdyt?

File: packages/core/src/api/exporters/markdown/__snapshots__/mention/basic/markdown.md [hunk 2]
```diff
@@ -0,0 +1 @@
+I enjoy working with\@Matthew
```
[the remark this answers is not in the document]
Comment (YousefED) [reply packages/core/src/api/exporters/markdown/__snapshots__/mention/basic/markdown.md:1]: Agree, will update this

File: packages/core/src/api/exporters/markdown/__snapshots__/hardbreak/start/markdown.md [hunk 3]
```diff
@@ -0,0 +1 @@
+Text1
```
[the remark this answers is not in the document]
Comment (YousefED) [reply packages/core/src/api/exporters/markdown/__snapshots__/hardbreak/start/markdown.md:1]: Good spot. My guess is that the markdown exporter collapses whitespace like this, and I think many editors do this. Imo let's keep it for now until someone thinks this is causing issues

File: packages/core/src/api/exporters/markdown/__snapshots__/simpleImage/button/markdown.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (YousefED) [reply packages/core/src/api/exporters/markdown/__snapshots__/simpleImage/button/markdown.md:1]: moved to https://github.com/TypeCellOS/BlockNote/issues/434, let's discuss there

[the remark this answers is not in the document]
Comment (matthewlipski) [reply packages/core/src/api/exporters/markdown/__snapshots__/simpleImage/button/markdown.md:1]: Tbh I prefer that approach of just unmounting the hidden elements, I think it's cleaner in general. Also I think it's better if the external html is identical to the rendered html (within `blockContainer` divs) - I can imagine a scenario where someone might be confused why their hidden elements aren't in the output, so we would have to document this additional processing step.

**Pull request merged**